esp/services/ws_configmgr/ws_configmgrService.cpp

```diff
@@ -593,6 +596,103 @@ bool Cws_configMgrEx::onFetchNodes(IEspContext &context, IEspFetchNodesRequest &
 }
 
 
+bool Cws_configMgrEx::onGetNodeCopy(IEspContext &context, IEspGetNodeCopyRequest &req, IEspGetNodeCopyResponse &resp)
+{
+    std::string nodeId = req.getNodeId();
+    std::string sessionId = req.getSessionId();
+    Status status;
+
+    ConfigMgrSession *pSession = getConfigSession(sessionId, true);
+
+    if (nodeId.empty())
+        nodeId = pSession->m_pEnvMgr->getRootNodeId();
```

Comment:
Can m_pEnvMgr be nullptr?

Reply:
No, provided 'true' was passed as the second parameter to getConfigSession. That indicates that a loaded environment is required.